pkg/pulumiyaml/run.go

```diff
@@ -911,16 +918,20 @@ func (e *programEvaluator) registerConfig(intm configNode) (interface{}, bool) {
 	if errors.Is(err, config.ErrMissingVar) && defaultValue != nil {
 		v = defaultValue
 	} else if err != nil {
-		return e.errorf(intm.Key, err.Error())
+		return e.errorf(intmKey, err.Error())
 	}
 
 	contract.Assertf(v != nil, "let an uninitialized var slip through")
 
 	// The value was marked secret in the configuration section, but in the
 	// config section. We need to wrap it in `pulumi.ToSecret`.
-	if (c.Secret != nil && c.Secret.Value) && !isSecretInConfig {
-		v = pulumi.ToSecret(v)
+	if intm, ok := intm.(configNodeYaml); ok {
+		c := intm.Value
+		if (c.Secret != nil && c.Secret.Value) && !isSecretInConfig {
+			v = pulumi.ToSecret(v)
+		}
 	}
+
```

Comment:
I think there is an env var that is used to mark `PULUMI_CONFIG` values as secret. We should address that here.

Reply:
I'm not quite sure if Friel wanted to add `PULUMI_CONFIG_SECRET_KEYS` (https://github.com/pulumi/pulumi-yaml/issues/369) or if it already exists because I can't find any documentation on it..